backend/middleware/auth.middleware.js

```diff
@@ -0,0 +1,66 @@
+const passport = require('passport');
+const passportLocal = require('passport-local');
+const passportJWT = require('passport-jwt');
+const bcrypt = require('bcrypt');
+
+const LocalStrategy = passportLocal.Strategy;
+const JWTStrategy = passportJWT.Strategy;
+const { ExtractJwt } = passportJWT;
+
+const db = require('../model/index');
+
+passport.use(new LocalStrategy(
+  {
+  usernameField: 'username',
+  passwordField: 'password',
+  },
+  async (username, password, done) => {
+    console.log('Authenticating login');
+
+    if(!username || !password){
+      return done(null, false, {message: "Username and password are required to login"});
+    }
+
+    try {
+      const user = await db.user.findOne({ where: { username: username }});
+
+      if(!user){
+        return done(null, false, { message: "No user found" });
+      }
+  
+      const matchingPassword = await bcrypt.compare(password, user.password);
+  
+      if(!matchingPassword) {
+        return done(null, false, { message: 'Password did not match user' });
+      }
+  
+      console.log('Login Successful. Logged in: ', user.username);
+
+      return done(null, user, { message: 'User logged in successfully' });
+    } catch (error) { return done(error);}
```

Comment:
OK. Close, but you went too far on the consolidation ... should be ...

```javascript
try {
 ...
} catch (error) {
  return done(error);
}
```

Reply:
Weird I saw it this way on my end but didn't have anything to commit. Fixed!